ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -99,6 +99,12 @@ public abstract void configureInputJobProperties(TableDesc tableDesc,
     Map<String, String> jobProperties);
 
   /**
+   * This method is called to allw the StorageHandlers the chance to
+   * populate secret keys into the job's credentials.
+   */
+  public abstract void configureInputJobCredentials(TableDesc tableDesc, Map<String, String> secrets);
```

Comment:
Do you think this is a concern? I could create a sub-interface. That's kind of lame, but won't require old handlers to change...